README.md

```diff
@@ -0,0 +1,223 @@
+
+## Steps
+
+The codemod executes a sequence of several ordered tasks. At the end of the process, reinstall your dependencies; if your app follows the standards the codemod expects, it should now builds correctly using Vite. If something went wrong, the sections below detail what the codemod expectations are and what it tries to output so you can figure out how to complete the migration manually.
+
+### Checking for unsupported dependencies
+
+This is a verification task, it doesn't change your code.
+
+The codemod will first check there are no known dependency that is incompatible with the way Vite works. `ember-fetch` needs to go.
+
+### Checking addons are v2
+
+This is a verification task, it doesn't change your code.
+
+The codemod will look at all your Ember dependencies and will advise you for updates.
+
+When you use Embroider in an app that depends on v1 addons, Embroider will try to auto-fix the v1 addons a way that make them compatible with Vite. This approach works for a number of known addons, but we cannot guarantee it will work for any v1 addon you use. The best way to avoid issues is that your classic app already relies only on v2 addons, and the codemod will guide you in that direction:
+
+- If one of your addons is v1 but the latest version on npm is v2, it's recommended to update.
+- If one of your addons is v1 and no v2 format is available, it's recommended to look for a different alternative or make the addon v2.
+- If one of your v1 addons but is known as being correctly rewriten by Embroider, the codemod won't notice you about it.
+- If one of your v1 addons comes from Ember app blueprint and is no longer used in Embroider+Vite world, the codemod won't notice you about it.
+
+### Creating new required files
+
+From this step, the codemod is done with verifications and starts modifying your code.
+
+First, it creates new files that are now required for Embroider+Vite. These files are copied direclty from the [embroider-build/app-blueprint](https://github.com/embroider-build/app-blueprint), so you can refer to this repository to see the expected content:
+
+- [app/config/environment.js](https://github.com/embroider-build/app-blueprint/blob/main/files-override/shared/app/config/environment.js)
+- [vite.config.mjs](https://github.com/embroider-build/app-blueprint/blob/main/files-override/shared/vite.config.mjs)
+- [babel.config.cjs](https://github.com/embroider-build/app-blueprint/blob/main/files/js/babel.config.cjs)
+
+### Moving `index.html` at the root
+
+Vite expects the `index.html` to be at the root of the app. The codemod will move your existing `app/index.html` at the root rather than creating a new file, this way it keeps all the customizations you added.
```

Comment:
```suggestion
Vite expects the `index.html` to be at the root of the app. The codemod will move your existing `app/index.html` to the root rather than creating a new file, this way it keeps all the customizations you added.
```